fragment.go

```diff
@@ -350,30 +350,41 @@ func (f *fragment) row(rowID uint64) *Row {
 	return f.unprotectedRow(rowID)
 }
 
+// unprotectedRow returns a row from the row cache if available or from storage
+// (updating the cache).
```

Comment:
I'm curious why this didn't previously grab rows from storage. Is this a performance improvement, or a bug fix, or ??.

Reply:
there's actually no functional change here. I broke out the part of this method that copied the row out of storage so that I could use it elsewhere.

I have some thoughts around getting rid of the rowCache entirely, but not fully baked yet.